homeassistant/components/smartthings/climate.py

```diff
@@ -462,3 +493,54 @@ def target_temperature(self):
     def temperature_unit(self):
         """Return the unit of measurement."""
         return UNIT_MAP.get(self._device.status.attributes[Attribute.temperature].unit)
+
+    @property
+    def swing_modes(self) -> list[str] | None:
+        """Return the list of available swing modes."""
+        supported_modes = self._device.status.attributes[
+            Attribute.supported_fan_oscillation_modes
+        ][0]
+        supported_swings = [
+            FAN_OSCILLATION_TO_SWING.get(m, SWING_OFF) for m in supported_modes
+        ]
+        return supported_swings
+
+    async def async_set_swing_mode(self, swing_mode: str) -> None:
+        """Set swing mode."""
+        fan_oscillation_mode = SWING_TO_FAN_OSCILLATION[swing_mode]
+        await self._device.set_fan_oscillation_mode(fan_oscillation_mode)
+
+        # setting the fan must reset the preset mode (it deactivates the windFree function)
+        self._attr_preset_mode = ""
+
+    @property
+    def swing_mode(self) -> str:
+        """Return the swing setting."""
+        return FAN_OSCILLATION_TO_SWING.get(
+            self._device.status.fan_oscillation_mode, SWING_OFF
+        )
+
+    @property
+    def preset_modes(self) -> list[str]:
```

Comment:
Would be better to "calc" this once during `__init__` so we can simply return if it's there or not